### ISSUE TITLE
Static members cause java.lang.ArrayIndexOutOfBoundsException:

### ISSUE THREAD

**Issue by shohamlevy** [strapdata/elassandra#57]
Using Elassandra 2.1.1-17

Creating a document with a static member, and trying to fetch data, causes the ArrayIndexOutOfBounds exception.

Example:

```
curl -XPUT "localhost:9200/test?pretty=true" -d '{
    "settings": {
        "keyspace":"test"
    },
    "mappings": {
            "t2": {
                "properties": {
                    "id": {"type": "integer", "cql_collection": "singleton", "index": "no", "cql_primary_key_order": 0,
                        "cql_partition_key": true},
                    "surname": {"type": "string", "cql_collection": "singleton", "index": "analyzed", "cql_primary_key_order": 1,
                        "cql_partition_key": true},
                    "name": {"type": "string", "cql_collection": "singleton", "index": "analyzed", "cql_primary_key_order": 2},
                    "phonetic_name": {"type": "string", "cql_collection": "singleton", "index": "not_analyzed", "cql_static_column": true},
                    "nicks": {"type": "string", "cql_collection": "list", "index": "analyzed"}
                }
            }
    }
}'
```

```
cqlsh> INSERT INTO test.t2 (id, surname, name, phonetic_name, nicks) VALUES (22, 'Genesis', 'Abraham', 'ai-b-ram', ['the-A', 'ab']);
```

**And then query:**

```
> curl -XGET "localhost:9200/test/t2/_search?pretty=true" -d '{ "query": { "matchAll": {}}}'

{
  "error" : {
    "root_cause" : [ {
      "type" : "array_index_out_of_bounds_exception",
      "reason" : "2"
    } ],
    "type" : "search_phase_execution_exception",
    "reason" : "all shards failed",
    "phase" : "query_fetch",
    "grouped" : true,
    "failed_shards" : [ {
      "shard" : 0,
      "index" : "test",
      "node" : "05a8fa11-0c44-46b9-97c0-d9e2f12fe481",
      "reason" : {
        "type" : "array_index_out_of_bounds_exception",
        "reason" : "2"
      }
    } ]
  },
  "status" : 500
}

```

**Stacktrace**

```
org.elasticsearch.action.search.SearchPhaseExecutionException: all shards failed
        at org.elasticsearch.action.search.type.TransportSearchTypeAction$BaseAsyncAction.onFirstPhaseResult(TransportSearchTypeAction.java:263) [elassandra-2.1.1-17.jar:na]
        at org.elasticsearch.action.search.type.TransportSearchTypeAction$BaseAsyncAction$1.onFailure(TransportSearchTypeAction.java:209) [elassandra-2.1.1-17.jar:na]
...
...
Caused by: java.lang.ArrayIndexOutOfBoundsException: 2
        at org.elasticsearch.cassandra.cluster.InternalCassandraClusterService.parseElasticId(InternalCassandraClusterService.java:1836) ~[elassandra-2.1.1-17.jar:na]
        at org.elasticsearch.cassandra.cluster.InternalCassandraClusterService.parseElasticId(InternalCassandraClusterService.java:1809) ~[elassandra-2.1.1-17.jar:na]
        at org.elasticsearch.search.fetch.FetchPhase.loadStoredFields(FetchPhase.java:425) ~[elassandra-2.1.1-17.jar:na]
        at org.elasticsearch.search.fetch.FetchPhase.createSearchHit(FetchPhase.java:212) ~[elassandra-2.1.1-17.jar:na]
        at org.elasticsearch.search.fetch.FetchPhase.execute(FetchPhase.java:177) ~[elassandra-2.1.1-17.jar:na]
        at org.elasticsearch.search.SearchService.executeFetchPhase(SearchService.java:504) ~[elassandra-2.1.1-17.jar:na]
        at org.elasticsearch.search.action.SearchServiceTransportAction$SearchQueryFetchTransportHandler.messageReceived(SearchServiceTransportAction.java:392) ~[elassandra-2.1.1-17.jar:na]
        at org.elasticsearch.search.action.SearchServiceTransportAction$SearchQueryFetchTransportHandler.messageReceived(SearchServiceTransportAction.java:1) ~[elassandra-2.1.1-17.jar:na]
        at org.elasticsearch.transport.TransportService$4.doRun(TransportService.java:350) [elassandra-2.1.1-17.jar:na]
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elassandra-2.1.1-17.jar:na]
        ... 3 common frames omitted
2016-10-30 11:40:26,940 INFO  [elasticsearch[localhost][listener][T#2]] BytesRestResponse.java:131 convert /test Params: {pretty=true, index=test}
org.elasticsearch.indices.IndexAlreadyExistsException: already exists
        at org.elasticsearch.cluster.metadata.MetaDataCreateIndexService.validateIndexName(MetaDataCreateIndexService.java:153) ~[elassandra-2.1.1-17.jar:na]
        at org.elasticsearch.cluster.metadata.MetaDataCreateIndexService.validate(MetaDataCreateIndexService.java:582) ~[elassandra-2.1.1-17.jar:na]
        at org.elasticsearch.cluster.metadata.MetaDataCreateIndexService.access$2(MetaDataCreateIndexService.java:581) ~[elassandra-2.1.1-17.jar:na]
        at org.elasticsearch.cluster.metadata.MetaDataCreateIndexService$2.execute(MetaDataCreateIndexService.java:262) ~[elassandra-2.1.1-17.jar:na]
        at org.elasticsearch.cluster.service.InternalClusterService$UpdateTask.run(InternalClusterService.java:473) ~[elassandra-2.1.1-17.jar:na]
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:231) ~[elassandra-2.1.1-17.jar:na]
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:194) ~[elassandra-2.1.1-17.jar:na]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_51]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_51]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_51]

```

**Comment by vroyer** [comment]
Yes there is a bug.
It’s fixed but i still need to run some automated non-regression tests, it should be available next week.
Thanks’ for reporting this issue.

**Comment by vroyer** [comment]
Fixed in release 2.1.1-18.1.

curl -XGET "localhost:9200/test/t2/_search?pretty=true" -d '{ "query": { "matchAll": {}}}'
{
  "took" : 64,
  "timed_out" : false,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "failed" : 0
  },
  "hits" : {
    "total" : 2,
    "max_score" : 1.0,
    "hits" : [ {
      "_index" : "test",
      "_type" : "t2",
      "_id" : "[22,\"Genesis\"]",
      "_score" : 1.0,
      "_source":{"phonetic_name":"ai-b-ram"}
    }, {
      "_index" : "test",
      "_type" : "t2",
      "_id" : "[22,\"Genesis\",\"Abraham\"]",
      "_score" : 1.0,
      "_source":{"phonetic_name":"ai-b-ram","surname":"Genesis","name":"Abraham","id":22,"nicks":["the-A","ab"]}
    } ]
  }
}

curl -XGET "localhost:9200/test/t2/_search?pretty&q=nicks:ab"
{
  "took" : 75,
  "timed_out" : false,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "failed" : 0
  },
  "hits" : {
    "total" : 1,
    "max_score" : 0.5,
    "hits" : [ {
      "_index" : "test",
      "_type" : "t2",
      "_id" : "[22,\"Genesis\",\"Abraham\"]",
      "_score" : 0.5,
      "_source":{"phonetic_name":"ai-b-ram","surname":"Genesis","name":"Abraham","id":22,"nicks":["the-A","ab"]}
    } ]
  }
}

$ curl -XGET "localhost:9200/test/t2/_search?pretty&q=phonetic_name:ai-b-ram"
{
  "took" : 5,
  "timed_out" : false,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "failed" : 0
  },
  "hits" : {
    "total" : 1,
    "max_score" : 1.6931472,
    "hits" : [ {
      "_index" : "test",
      "_type" : "t2",
      "_id" : "[22,\"Genesis\"]",
      "_score" : 1.6931472,
      "_source":{"phonetic_name":"ai-b-ram"}
    } ]
  }
}

**Comment by shohamlevy** [comment]
Tested it. Works fine.

Thanks!
